### ISSUE TITLE
[Crash]: Ender Dragon Attack (1.16.5 36.2.39)

### ISSUE THREAD

**Issue by dayd21git** [Yesssssman/epicfightmod#1350]
### Have you checked if a similar issue is already reported by someone else?

- [X] I checked there are no similar issues have been reported.

### Have you read the support policy?

- [X] I read it and I accept the policy.

### Are you using the latest Epic Fight and recommended Forge version?

- [X] I checked I'm using latest Epic Fight and recommended Forge version.

### The mod

https://legacy.curseforge.com/minecraft/mc-mods/epic-fight-mod/files/4029362

### Minecraft Version

1.16.5

### What happened?
```
---- Minecraft Crash Report ----
// Oops.

Time: 7/16/23 3:21 PM
Description: Ticking entity

java.lang.NullPointerException: Ticking entity
	at yesman.epicfight.world.capabilities.entitypatch.boss.enderdragon.DragonFlyingPhase.findNewTarget(DragonFlyingPhase.java:60) ~[epicfight:16.6.4] {re:classloading}
	at yesman.epicfight.world.capabilities.entitypatch.boss.enderdragon.DragonFlyingPhase.func_188659_c(DragonFlyingPhase.java:35) ~[epicfight:16.6.4] {re:classloading}
	at net.minecraft.entity.boss.dragon.EnderDragonEntity.func_70636_d(EnderDragonEntity.java:229) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.entity.LivingEntity.func_70071_h_(LivingEntity.java:2623) ~[?:?] {re:mixin,pl:accesstransformer:B,xf:fml:better_diving:LivingEntity canBreatheUnderwater Transformer,xf:fml:better_diving:LivingEntity decreaseAirSupply Transformer,xf:fml:better_diving:LivingEntity increaseAirSupply Transformer,xf:fml:placebo:placeboshieldblock,xf:fml:better_diving:LivingEntity travel Transformer,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:better_diving:LivingEntity canBreatheUnderwater Transformer,xf:fml:better_diving:LivingEntity decreaseAirSupply Transformer,xf:fml:better_diving:LivingEntity increaseAirSupply Transformer,xf:fml:placebo:placeboshieldblock,xf:fml:better_diving:LivingEntity travel Transformer,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:cgm.mixins.json:common.LivingEntityMixin,pl:mixin:APP:upgrade_aquatic.mixins.json:LivingEntityMixin,pl:mixin:APP:mixins.timecore.json:MixinLivingEntity,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.entity.MobEntity.func_70071_h_(MobEntity.java:343) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betteranimalsplus.mixins.json:MobAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ServerWorld.func_217479_a(ServerWorld.java:756) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:starlight.mixins.json:common.world.ServerWorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.World.func_217390_a(World.java:811) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:starlight.mixins.json:common.world.WorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ServerWorld.func_72835_b(ServerWorld.java:529) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:starlight.mixins.json:common.world.ServerWorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:1049) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:336) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:943) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:775) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:265) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at yesman.epicfight.world.capabilities.entitypatch.boss.enderdragon.DragonFlyingPhase.findNewTarget(DragonFlyingPhase.java:60) ~[epicfight:16.6.4] {re:classloading}
	at yesman.epicfight.world.capabilities.entitypatch.boss.enderdragon.DragonFlyingPhase.func_188659_c(DragonFlyingPhase.java:35) ~[epicfight:16.6.4] {re:classloading}
	at net.minecraft.entity.boss.dragon.EnderDragonEntity.func_70636_d(EnderDragonEntity.java:229) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.entity.LivingEntity.func_70071_h_(LivingEntity.java:2623) ~[?:?] {re:mixin,pl:accesstransformer:B,xf:fml:better_diving:LivingEntity canBreatheUnderwater Transformer,xf:fml:better_diving:LivingEntity decreaseAirSupply Transformer,xf:fml:better_diving:LivingEntity increaseAirSupply Transformer,xf:fml:placebo:placeboshieldblock,xf:fml:better_diving:LivingEntity travel Transformer,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:better_diving:LivingEntity canBreatheUnderwater Transformer,xf:fml:better_diving:LivingEntity decreaseAirSupply Transformer,xf:fml:better_diving:LivingEntity increaseAirSupply Transformer,xf:fml:placebo:placeboshieldblock,xf:fml:better_diving:LivingEntity travel Transformer,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:cgm.mixins.json:common.LivingEntityMixin,pl:mixin:APP:upgrade_aquatic.mixins.json:LivingEntityMixin,pl:mixin:APP:mixins.timecore.json:MixinLivingEntity,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.entity.MobEntity.func_70071_h_(MobEntity.java:343) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betteranimalsplus.mixins.json:MobAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ServerWorld.func_217479_a(ServerWorld.java:756) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:starlight.mixins.json:common.world.ServerWorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
-- Entity being ticked --
Details:
	Entity Type: minecraft:ender_dragon (net.minecraft.entity.boss.dragon.EnderDragonEntity)
	Entity ID: 852
	Entity Name: Ender Dragon
	Entity's Exact location: -33.40, 186.19, -34.76
	Entity's Block location: World: (-34,186,-35), Chunk: (at 14,11,13 in -3,-3; contains blocks -48,0,-48 to -33,255,-33), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Entity's Momentum: 0.28, 0.05, 0.79
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null
Stacktrace:
	at net.minecraft.world.World.func_217390_a(World.java:811) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:starlight.mixins.json:common.world.WorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.world.server.ServerWorld.func_72835_b(ServerWorld.java:529) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:starlight.mixins.json:common.world.ServerWorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}


-- Affected level --
Details:
	All players: 2 total; [ServerPlayerEntity['_Trinadsatiy_'/284, l='ServerLevel[timerp]', x=-56.58, y=171.85, z=-48.50](TextComponent{text='_Trinadsatiy_', siblings=[], style=Style{ color=null, bold=null, italic=null, underlined=null, strikethrough=null, obfuscated=null, clickEvent=null, hoverEvent=null, insertion=null, font=minecraft:default}} at -56.57739970467449,171.851133345205,-48.49890672686445), ServerPlayerEntity['day-d'/369, l='ServerLevel[timerp]', x=-122.72, y=91.97, z=-111.16](TextComponent{text='day-d', siblings=[], style=Style{ color=null, bold=null, italic=null, underlined=null, strikethrough=null, obfuscated=null, clickEvent=null, hoverEvent=null, insertion=null, font=minecraft:default}} at -122.72186722892539,91.97446171627455,-111.15864218785195)]
	Chunk stats: ServerChunkCache: 3920
	Level dimension: minecraft:timerp
	Level spawn location: World: (81,77,144), Chunk: (at 1,4,0 in 5,9; contains blocks 80,0,144 to 95,255,159), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 116763239 game time, 6322376 day time
	Level name: timerp
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Level weather: Rain time: 9824 (now: false), thunder time: 35336 (now: false)
	Known server brands: forge
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:1049) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:336) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:943) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:775) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:265) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292] {}


-- System Details --
Details:
	Minecraft Version: 1.16.5
	Minecraft Version ID: 1.16.5
	Operating System: Linux (amd64) version 5.10.0-21-amd64
	Java Version: 1.8.0_292, AdoptOpenJDK
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), AdoptOpenJDK
	Memory: 583628672 bytes (556 MB) / 1689063424 bytes (1610 MB) up to 2138767360 bytes (2039 MB)
	CPUs: 1
	ModLauncher: 8.1.3+8.1.3+main-8.1.x.c94d18ec
	ModLauncher launch target: fmlserver
	ModLauncher naming: srg
	ModLauncher services: 
		/mixin-0.8.4.jar mixin PLUGINSERVICE 
		/eventbus-4.0.0.jar eventbus PLUGINSERVICE 
		/server.jar object_holder_definalize PLUGINSERVICE 
		/server.jar runtime_enum_extender PLUGINSERVICE 
		/accesstransformers-3.0.1.jar accesstransformer PLUGINSERVICE 
		/server.jar capability_inject_definalize PLUGINSERVICE 
		/server.jar runtimedistcleaner PLUGINSERVICE 
		/mixin-0.8.4.jar mixin TRANSFORMATIONSERVICE 
		/server.jar fml TRANSFORMATIONSERVICE 
	FML: 36.2
	Forge: net.minecraftforge:36.2.39
	FML Language Providers: 
		javafml@36.2
		minecraft@1
```

**Comment by MetalKnight56** [comment]
Closing issue as 1.16.5 is no longer supported.

**Comment by dayd21git** [comment]
> Closing issue as 1.16.5 is no longer supported.

You have published version 1.16.5, but it does not work correctly. No one is asking you to introduce new features into this development, just fix the problem or remove your incorrect work from public access!

**Comment by MetalKnight56** [comment]
> > Closing issue as 1.16.5 is no longer supported.
> 
> You have published version 1.16.5, but it does not work correctly. No one is asking you to introduce new features into this development, just fix the problem or remove your incorrect mod(version) from public access!

Shut up, we said it countless times its no longer supported, if you want it fixed, do it yourself, pull requests exists you moron.
Again we are not forcing you to play the goddam mod, its public for anyone willing to use a non longer supported version, face your consequences. and don't even try and open more issues here, or else i'll block you from this repo.

**Comment by dayd21git** [comment]
> > > Closing issue as 1.16.5 is no longer supported.
> > 
> > 
> > You have published version 1.16.5, but it does not work correctly. No one is asking you to introduce new features into this development, just fix the problem or remove your incorrect mod(version) from public access!
> 
> Shut up, we said countless times its no longer supported, if you want it fixed, do it yourself, pull requests exists you moron. Again we are not forcing you to play the goddam mod, its public for anyone willing to use a non longer supported version, face your consequences. and don't even try and open more issues here, or else i'll block you from this repo.

The modification has a release status, which implies the full operation of the mod. If I knew the mod had bugs, I wouldn't have downloaded it. You should move the modification status to beta, because it leads to crashes.

And since you have switched to insults, there is no point in talking to you anymore. I will discuss this issue with the support of github and curseforge.
